src/Encoder.js

```diff
@@ -50,6 +58,13 @@ class Encoder {
     */
     encode(contract, funName, args) {
         const argTypes = this.#typeResolver.getCallTypes(contract, funName)
+
+        if (args.length !== argTypes.length) {
```

Comment:
why don't allow to omit optional arguments at the end as we have it now? This seems to be native to JavaScript. This change is a breaking one.

So, in TypeScript it would correspond only to `v: <type> | undefined`, won't be compatible with `v?: type` anymore.

Reply:
Because I don't have good ideas how to implement it. Mixing code/class responsibility (as it was before) is no go.

Reply:
I might have found a acceptable implementation

Reply:
Pushed. Let me know if that works for you.